ElectronClient/app/gui/ResourceScreen.tsx

```diff
@@ -0,0 +1,180 @@
+	orphaned: (resource: IResource) =>
+		resource.isOrphan ? 0 : 1,
+	size: (resource: IResource) =>
+		1 / resource.daoResource.size,
+	name: (resource: IResource) =>
+		resource.title,
+};
+
+const getIteratees = (sortingOrders: SortingOrder[]) => {
+	const ordering = (Object.keys(sortIteratees) as SortingOrder[])
+		.filter(order => sortingOrders.includes(order));
+	return ordering.map(order => sortIteratees[order]);
+};
+
+const isOrphan = (noteResources: any[] | undefined): boolean => {
+	if (!noteResources) {
+		return false;
+	}
+	return noteResources.filter((nr) => nr.is_associated === 1).length === 0;
+};
+
+const buildResources = (resources: any[], noteResources: any[]) => {
+	const groupedNoteResources = groupBy(noteResources, 'resource_id');
+	return resources.map((resource: any) => ({
+		title: resource.title,
+		id: resource.id,
+		isOrphan: isOrphan(groupedNoteResources[resource.id]),
+		daoResource: resource,
+		size: prettyBytes(resource.size),
+	} as IResource));
+};
+
+const ResourceTable: React.FC<IResourceTable> = (props: IResourceTable) => {
+	const sortedResources = orderBy([ ...props.resources ], getIteratees(props.sortingOrders));
+	const sortOrderEngagedMarker = (s: SortingOrder) => (
+		<a href="#" onClick={ () => props.onToggleSortOrder(s) }>{ props.sortingOrders.includes(s) ? '▾' : '▴' }</a>
+	);
+	return <table style={{ width: '90%' }}>
+		<thead>
+			<tr>
+				<th>Title { sortOrderEngagedMarker('name') }</th>
+				<th>Size { sortOrderEngagedMarker('size') }</th>
+				<th>ID</th>
+				<th>Action</th>
+			</tr>
+		</thead>
+		<tbody>
+			{sortedResources.map((resource: IResource, index: number) =>
+				<tr key={index}>
+					<td>
+						<a href="#" onClick={() => props.onResourceClick(resource)}>{resource.title}</a>
+						{ resource.isOrphan && <i>(Orphan)</i> }
+					</td>
+					<td>{resource.size}</td>
+					<td>{resource.id}</td>
+					<td>
+						<button onClick={ () => props.onResourceDelete(resource) }>Delete</button>
+					</td>
+				</tr>
+			)}
+		</tbody>
+	</table>;
+};
+
+const openResource = (resource: IResource) => {
+	const resourcePath = Resource.fullPath(resource.daoResource);
+	const ok = bridge().openExternal(`file://${resourcePath}`);
+	if (!ok) {
+		bridge().showErrorMessageBox(_('This file could not be opened: %s', resourcePath));
+	}
+};
+
+const deleteResource = (resource: IResource): Promise<any> => Resource.batchDelete([resource.id]);
+
+class ResourceScreenComponent extends React.Component<IProps, IState> {
+	constructor(props: IProps) {
+		super(props);
+		this.state = {
+			resources: undefined,
+			sortingOrders: ['size', 'orphaned'],
+		};
+	}
+
+	async reloadResources() {
+		const noteResources = await NoteResource.modelSelectAll('SELECT * FROM note_resources');
+		const resources = await Resource.modelSelectAll('SELECT * FROM resources');
+		this.setState({ resources: buildResources(resources, noteResources) });
```

Comment:
If you're going to load all the resources, which maybe is fine, you need at least to select only the columns that you need, such as id, title, etc.